src/solve.jl

```diff
@@ -75,7 +75,7 @@ function init(prob::AbstractDDEProblem{uType,tType,lType,isinplace}, alg::algTyp
                                                        ode_prob,
                                                        OrdinaryDiffEq.alg_order(alg)))
     end
-    integrator.dt = dt
+    integrator.dt = zero(dt)
```

Comment:
This specific line here is without any practical impact, I would say. For a new initialised DDE integrator `dde_int` it leads to `isnan.(dde_int.f(t, u))` for any `t > minimum(prob.lags)` since any interpolation `current_interpolant` is also not a number - but it does not influence the calculations since the initial step is always at most of size of the minimal delay (assured since the discontinuities are tstops).

However, for all other steps it makes a huge difference (by the addition of these updates I got rid of the exploding fixed-point iterations in some tests!), since any interpolation that is used in the function evaluation and hence also the calculation of the next steps is based on the current time interval of the ODE integrator, and in particular also on the current value of `dt` (see https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/dense/generic_dense.jl#L40 and similar lines). These lines also show that for an initial dt of zero `Θ` is calculated by a division by zero, and hence this explains why the interpolation does not work initially.

So this specific line could be removed but I thought it would be preferable to keep `dt`, `t`, and `tprev` synchronized all the time.